### DISCUSSION TITLE
Generalized way to process macros that produce const values

### DISCUSSION THREAD

**Discussion by corecode** [rust-lang/rust-bindgen#2649] [Ideas]
It seems that we could obtain the const values of macros by passing them through clang parsing.  For this we'd have to collect all macros (we already sort of do so), then create a temporary file to run through clang again.  This file could contain constructs like

```
const typeof(MYMACRO) MYMACRO_value = MYMACRO;
```

Does this seems like a viable path to turn all kinds of macros into const values?

**Comment by pvdrz** [comment]
There's a similar discussion happening here: https://github.com/rust-lang/rust-bindgen/issues/753
